drivers/sensor/iis2mdc/iis2mdc.h

```diff
@@ -11,40 +11,39 @@
 #ifndef __MAG_IIS2MDC_H
 #define __MAG_IIS2MDC_H
 
-#include <drivers/spi.h>
 #include <drivers/gpio.h>
 #include <drivers/sensor.h>
 #include <sys/util.h>
 #include "iis2mdc_reg.h"
 
-union axis3bit16_t {
-	int16_t i16bit[3];
-	uint8_t u8bit[6];
-};
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#include <drivers/spi.h>
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
 
-union axis1bit16_t {
-	int16_t i16bit;
-	uint8_t u8bit[2];
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+#include <drivers/i2c.h>
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c) */
+
+
+union iis2mdc_bus_cfg {
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+	uint16_t i2c_slv_addr;
+#endif
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+	struct spi_config spi_cfg;
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
 };
 
-struct iis2mdc_config {
-	char *master_dev_name;
+struct iis2mdc_dev_config {
+	const struct device *bus;
 	int (*bus_init)(const struct device *dev);
+	const union iis2mdc_bus_cfg bus_cfg;
 #ifdef CONFIG_IIS2MDC_TRIGGER
 	const char *drdy_port;
 	gpio_pin_t drdy_pin;
 	gpio_dt_flags_t drdy_flags;
```

Comment:
Can just make this a `struct gpio_dt_spec` instead when applying the above suggestion.

Reply:
Yes, done both! Thanks